hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -8524,6 +8530,39 @@ void checkAccess(String src, FsAction mode) throws IOException {
     logAuditEvent(true, operationName, src);
   }
 
+  /**
+   * Check if snapshot roots are created for all existing snapshottable
+   * directories. Create them if not.
+   */
+  void checkAndProvisionSnapshotTrashRoots() throws IOException {
+    if (haEnabled) {
+      if (!inActiveState()) {
```

Comment:
I am not 100% sure about this condition check. Any suggestions/confirmations?

The goal is to only let **Active NN** to check and provision snapshot trash roots. The assumption is that the `mkdirs()` call below propagates the write to standby NameNode.